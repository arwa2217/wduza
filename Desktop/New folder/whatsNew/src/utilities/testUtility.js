import React from "react";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import initialAppState from "../store/initialAppState";

const mockStore = configureMockStore([thunk]);

const store = mockStore(initialAppState);

export default {
  mount: (Component, props = {}, localState) => {
    let localStore = store;
    if (localState) {
      localStore = mockStore(localState);
    }
    const wrapper = mount(
      <Provider store={localStore}>
        <Component {...props} />
      </Provider>
    );
    return wrapper.find(Component).first();
  },
};
