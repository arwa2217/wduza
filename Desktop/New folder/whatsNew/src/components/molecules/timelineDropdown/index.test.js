import React from "react";
import testUtility from "@utilities/testUtility";
import TimelineDropdown from "./index";
import { act } from "react-dom/test-utils";

let timelineData = {
  currentOption: "Thursday, May 7th",
  options: [
    { text: "Yesterday" },
    { text: "Last Week" },
    { text: "Last month" },
    { text: "Last Year" },
    { text: "The very beginning" },
    { text: "Most Recent" },
    {
      text: "Time zone, switch",
      subNav: true,
      options: [{ text: "Option1" }, { text: "Option2" }],
    },
  ],
  navigateTo: jest.fn(),
};

describe("Timeline dropdown", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = testUtility.mount(TimelineDropdown, timelineData);
    wrapper.props();
  });
  it("should match the snap shot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("'s button should match text'", () => {
    expect(wrapper.find("button").text()).toEqual("Thursday, May 7th");
  });

  it("should have 8 child", () => {
    expect(wrapper.find("li").length).toEqual(8);
  });

  it("should call navigateTo on li click", () => {
    wrapper.find("li").at(1).simulate("click");
    expect(timelineData.navigateTo).toHaveBeenCalled();
  });
});
