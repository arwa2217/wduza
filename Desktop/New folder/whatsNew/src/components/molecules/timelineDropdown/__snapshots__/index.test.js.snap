// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline dropdown should match the snap shot 1`] = `
<_default
  currentOption="Thursday, May 7th"
  navigateTo={[MockFunction]}
  options={
    Array [
      Object {
        "text": "Yesterday",
      },
      Object {
        "text": "Last Week",
      },
      Object {
        "text": "Last month",
      },
      Object {
        "text": "Last Year",
      },
      Object {
        "text": "The very beginning",
      },
      Object {
        "text": "Most Recent",
      },
      Object {
        "options": Array [
          Object {
            "text": "Option1",
          },
          Object {
            "text": "Option2",
          },
        ],
        "subNav": true,
        "text": "Time zone, switch",
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-AxjAm eUxFNP"
    >
      <button
        onClick={[Function]}
      >
        Thursday, May 7th
      </button>
      <ul
        className=""
      >
        <li>
          Jump to...
        </li>
        <li
          className=""
          key="Yesterday0"
          onClick={[Function]}
        >
          Yesterday
        </li>
        <li
          className=""
          key="Last Week1"
          onClick={[Function]}
        >
          Last Week
        </li>
        <li
          className=""
          key="Last month2"
          onClick={[Function]}
        >
          Last month
        </li>
        <li
          className=""
          key="Last Year3"
          onClick={[Function]}
        >
          Last Year
        </li>
        <li
          className=""
          key="The very beginning4"
          onClick={[Function]}
        >
          The very beginning
        </li>
        <li
          className=""
          key="Most Recent5"
          onClick={[Function]}
        >
          Most Recent
        </li>
        <li
          className="sub-nav"
          key="Time zone, switch6"
          onClick={[Function]}
        >
          Time zone, switch
          <Icon
            as="i"
            name="chevron right"
          >
            <i
              aria-hidden="true"
              className="chevron right icon"
              onClick={[Function]}
            />
          </Icon>
        </li>
      </ul>
    </div>
  </styled.div>
</_default>
`;
